### ISSUE TITLE
changing back to old job names

### ISSUE THREAD

**Issue by dgordon562** [PacificBiosciences/FALCON#443]
Hi, Chris,

Would it be possible to return to the older naming of jobs so that I can see at a glance in qstat whether a job is a daligner job, an LAsort/merge job, a pread daligner job, etc?

I would prefer to have this information at a glance, without having to qstat -j, grep for script_file, etc.

I don't know by looking what kind of a job this is:

Jcc172b9d8

But in the older system this tells me a lot:

rj_12b7.sh

Not only does it tell me this is a daligner job, but it also tells me which job_ subdirectory it is working out of.

Can you see this information at a glance at Jcc172b9d8 ?  

Thanks!
David

**Comment by pb-cdunn** [comment]
Without unique job-names, we would need to record somehow the jobids from the `qsub` commands. (I hope that's obvious.) That's quite a bit of work.

Also, I'm told that we need uniqueness in the first 15 letters for the sake of some job-dist system (PBS maybe).

Without running `qstat`, you can look into the `mypwatcher/` directory to see what each job is actually doing. Is that sufficient?

**Comment by dgordon562** [comment]
No, looking in mypwatcher doesn't appear to be sufficient for a number of reasons:  1) if the job is in qw state, there is nothing yet in that directory and 2) if you are on the head node you don't have access to this directory.

The way it was before, qstat told you at a GLANCE how many jobs were running of what type.  Now it is a 3 step procedure for every single job:  qstat -j (job_number), look at script name, cat script.  PITA and not easily susceptible to looking at *all* jobs. 

  If uniqueness is the issue have the job name be:  (s)(dir)_(sequential#) where dir would be job_12, sequential # would be for uniqueness, and (s) would be a single character indicating whether it was a raw_reads or preads job.  That all fits in 13 characters.  This is just after thinking about it for 5 seconds, and I'm sure you could do better.  The point is that one should be able to know the type of job and the directory by looking at the first 10 (ten) characters of the job name since that is what is displayed in lists of jobs.

**Comment by pb-cdunn** [comment]
Hmmm. I *strongly* prefer longer, descriptive job-names, but then we would use different job-names for different systems. The 15-character limit is real for some systems.

>  That all fits in 13 characters.

For 2 different users on the same host? That's the problem.

Why not just tail the qsub commands into a file? We could do that trivially, and that should tell you everything you need to know. Would that suffice?

**Comment by dgordon562** [comment]
It might work.  But why keep trying different possibilities when we have one that we know will work?  What if you go to the work of putting the qsub commands into a file and it still doesn't satisfy us?  Then you try something else?  Just put the important info into the first part of the job name and it is done.

You can make the job names as long as you like (e.g., appending usernames or whatever)--just as long as you put the info we want into the first part of the job name.   Different users on the same system with the same job name is not a problem since sge distinguishes them.  I generally only ask sge to list my jobs--not others.

**Comment by pb-cdunn** [comment]
The job-names cannot be longer than 15 characters, and they must be globally unique with respect to different runs by different users within a system.

But dumping a simple file is simple. If you need more info there, we can dump more info. No problem.
